server/middleware/session/setSession.js

```diff
@@ -75,8 +77,26 @@ const setSession = async (req, res, next) => {
   } catch (e) {
     // console.error('Jwt Verify failed', e)
 
-    if (e.name === 'TokenExpiredError') {
-      res.clearCookie('idToken')
+    if (e instanceof TokenExpiredError) {
+      // Don't redirect user to an /api/ path after signing through,
+      // the user would be treated with a nice wall of json.
+      // This should only happen if the access token expires after page load but before ajax is exectued on a page.
+      const redirectUrl = req.url.startsWith('/api/') ? '/home/' : req.url
```

Comment:
Won't this result in API requests made from the client (after the token has expired) resulting in a 200 response that contains the home page HTML?

Would it be better in the event we find the token has expired and the current request is to the API to give a response that the client would be able to identify as such and handle appropriately (e.g. send a 401 status code)?

Reply:
My bad, I think you are partially correct. By default I don't think the redirect will be followed with fetch. It depends on what HTTP abstraction your using in the browser.  xmlhttprequest will follow redirects but fetch doesn't. 

All that would be needed to make this work would be to add "redirect:follow" to the fetch object on the front end https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Supplying_request_options

A 401 would be more appropriate. However it will need to be handled client side. In my manual testing it appeared the requests just fail. (I haven't looked at the code yet)

Reply:
Hi James, I have changed the code to return a 401 status code. I'm not sure what the client side behavior will be.